pkgs/top-level/julia-packages.nix

```diff
@@ -0,0 +1,60 @@
+{ pkgs
+, lib
+, stdenv
+, newScope
+, julia
+}:
+
+with lib;
+
+makeScope newScope (self:
+  let
+    callPackage = self.callPackage;
+
+    buildJuliaPackage = callPackage ../development/compilers/julia/build-julia-package.nix {
+      inherit lib stdenv;
+      inherit julia;
+      inherit computeRequiredJuliaPackages computeJuliaDepotPath computeJuliaLoadPath computeJuliaArtifacts;
+    };
+
+    # Given a list of required Julia package derivations, get a list of
+    # ALL required Julia packages needed for the ones specified to run.
+    computeRequiredJuliaPackages = with lib; packages:
+      let
+        hasJuliaPackage = drv: drv ? isJuliaPackage;
+        dependencies = packages':
+          unique (flatten (catAttrs "requiredJuliaPackages" packages')
+                  ++ (foldr (p: a: a ++ (dependencies p.requiredJuliaPackages)) [] (filter hasJuliaPackage  packages')));
+      in unique packages ++ (dependencies packages);
```

Comment:
This function of this is oddly identical to `propagatedBuildInputs`, maybe we really should copy what python is doing now.

Reply:
I'm not familiar with how Python distributes its packages. Can you make a concrete proposal?

Reply:
Just noticed that python is actually doing the same thing: requiredPythonModules, just their implementation appears to be more polished: https://github.com/NixOS/nixpkgs/blob/6fecd5a257327db6116b28acdfb45561e65e3dfa/pkgs/development/interpreters/python/python-packages-base.nix#L50

Reply:
The functions are not the same. My version includes all dependencies recursively, while the Python one only goes down one level.